### PR TITLE
777 introduce dynamic circuit inputs parsing for current e2e use case

### PR DESCRIPTION
**Pull request by ognjenkurtic**
# Description

Transaction agent executeTransaction method is refactored to translate transaction payload to circuit inputs using the schema previously associated with the workstep. It utilizes the circuitInputParser service for that.

Transaction agent executeTransaction method is also refactored for better readability.

CircuitInputParser service and schema is extended to support mapping  inputs from arrays of objects from the tx payload .

E2E test is extended to verify the flow works.

## Related Issue

#777 

## Motivation and Context

SRI needs to be able to support generic use-cases when it comes to circuits and this implementation allows for that.

## How Has This Been Tested

Extended unit tests
Extended e2e test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

File: examples/bri-3/src/bri/transactions/agents/transactions.agent.ts [hunk 1]
```diff
@@ -299,4 +291,86 @@ export class TransactionAgent {
...
+      `${process.env.MERKLE_TREE_HASH_ALGH}`,
+    );
+
+    const merkelizedInvoiceRoot = merkelizedPayload.getRoot().toString('hex');
+    const witnessHash = hashFn(JSON.stringify(witness)).toString('hex');
+
+    return hashFn(`${merkelizedInvoiceRoot}${witnessHash}`).toString('hex');
+  }
 }
+// TODO: Example input preparation for other workstep circuits from the example use-case, to be used
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/transactions/agents/transactions.agent.ts:346]: @ognjenkurtic is that gonna be an issue you create?

File: examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.spec.ts [hunk 2]
```diff
@@ -18,17 +18,12 @@ import {
 } from '../../../../identity/bpiSubjects/models/publicKey';
 
 jest.setTimeout(20000);
+//NOTE: Skiping out the workstep1 as it requires compiled artifacts to run.
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.spec.ts:21]: @ognjenkurtic will that be a another issue?

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM @ognjenkurtic ... check my comments

File: examples/bri-3/src/bri/transactions/agents/transactions.agent.ts (hunk 1 above)
Comment (ognjenkurtic) [reply examples/bri-3/src/bri/transactions/agents/transactions.agent.ts:346]: https://github.com/ethereum-oasis-op/baseline/issues/804

File: examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.spec.ts (hunk 2 above)
Comment (ognjenkurtic) [reply examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.spec.ts:21]: This was done by @biscuitdey . I believe these test are not needed as we can't expect the unit test suite to compile cirucits. Is it ok to remove these tests ? @biscuitdey

**Pull request merged**